connectors/tests/test_utils.py

```diff
@@ -51,13 +53,65 @@ def test_invalid_names():
             validate_index_name(name)
 
 
+@pytest.mark.asyncio
+async def test_mem_queue_speed(patch_logger):
+    # with memqueue
+    queue = MemQueue(maxmemsize=1024 * 1024, refresh_interval=0.1, refresh_timeout=2)
+    start = time.time()
+    for i in range(1000):
+        await queue.put("x" * 100)
+    mem_queue_duration = time.time() - start
+
+    # vanilla queue
+    queue = asyncio.Queue()
+    start = time.time()
+    for i in range(1000):
+        await queue.put("x" * 100)
+    queue_duration = time.time() - start
+
+    # mem queue should be 30 times slower at the most
+    quotient, _ = divmod(mem_queue_duration, queue_duration)
+    assert quotient < 30
```

Comment:
Why 30? :)

Reply:
Arbitrary number. It's just set to make sure the class does not get slower over time when we change it